contracts/ronin/validator/CoinbaseExecution.sol

```diff
@@ -112,36 +111,43 @@ abstract contract CoinbaseExecution is
    * @inheritdoc ICoinbaseExecution
    */
   function wrapUpEpoch() external payable virtual override onlyCoinbase whenEpochEnding oncePerEpoch {
-    uint256 newPeriod = _computePeriod(block.timestamp);
-    bool periodEnding = _isPeriodEnding(newPeriod);
-
-    address[] memory currValidatorIds = getValidatorIds();
-    address[] memory revokedCandidateIds;
-    uint256 epoch = epochOf(block.number);
-    uint256 nextEpoch = epoch + 1;
-    uint256 lastPeriod = currentPeriod();
-
-    _syncFastFinalityReward(epoch, currValidatorIds);
-
-    if (periodEnding) {
-      (uint256 totalDelegatingReward, uint256[] memory delegatingRewards) =
-        _distributeRewardToTreasuriesAndCalculateTotalDelegatingReward(lastPeriod, currValidatorIds);
-      _settleAndTransferDelegatingRewards(lastPeriod, currValidatorIds, totalDelegatingReward, delegatingRewards);
-      _tryRecycleLockedFundsFromEmergencyExits();
-      _recycleDeprecatedRewards();
-
-      ISlashIndicator slashIndicatorContract = ISlashIndicator(getContract(ContractType.SLASH_INDICATOR));
-      slashIndicatorContract.execUpdateCreditScores(currValidatorIds, lastPeriod);
-      (currValidatorIds, revokedCandidateIds) = _syncValidatorSet(newPeriod);
-      if (revokedCandidateIds.length > 0) {
-        slashIndicatorContract.execResetCreditScores(revokedCandidateIds);
+    unchecked {
+      uint256 newPeriod = _computePeriod(block.timestamp);
+      bool periodEnding = _isPeriodEnding(newPeriod);
```

Comment:
Please help tracking the `endBlock` for each period. If not increase size that much, tracking the `endBlock` for each epoch as well.